### PR TITLE
Adding ISO `Date` and `DateTime` memory benchmark

### PR DESCRIPTION
**Pull request by andrewpollack**
* Converting ISO dates into Calendar `Date`s via `Date::new_iso_date_from_integers`
* Converting ISO datetimes into Calendar `DateTime`s via `Date::new_iso_datetime_from_integers`
* Conditionally print results

### CODE REVIEW

**Comment by Manishearth** [comment]
I'll let @sffc (or @zbraniecki, if he's able) look at this: the benchmark looks good to me; however I have not spent as much time writing ICU4X benchmarks and I'll defer to them on this.

**Review by Manishearth — DISMISSED** [review]

File: components/calendar/examples/iso_date_manipulations.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        .collect::<Result<Vec<Date<Iso>>, _>>()
+        .expect("Failed to parse dates.");
+
+    for date in dates.iter() {
+        let formatted_date = format!(
+            "Year: {}, Month: {}, Day: {}",
+            date.year().number,
+            date.month().number,
+            date.day_of_month().0
+        );
```
Comment (zbraniecki) [inline components/calendar/examples/iso_date_manipulations.rs:56]: I think this should be part of `print` optional to printing. Memory used to allocate the string is not part of the cost we're measuring here.

Suggestion: move this to `print` behind the cfg.

File: components/calendar/examples/iso_datetime_manipulations.rs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    for datetime in datetimes.iter() {
+        let formatted_datetime = format!(
+            "Year: {}, Month: {}, Day: {}, Hour: {}, Minute: {}, Second: {}",
+            datetime.date.year().number,
+            datetime.date.month().number,
+            datetime.date.day_of_month().0,
+            u8::from(datetime.time.hour),
+            u8::from(datetime.time.minute),
+            u8::from(datetime.time.second),
```
Comment (zbraniecki) [inline components/calendar/examples/iso_datetime_manipulations.rs:58]: same here - make `print` accept the struct that has the right components, and conditionally print them. This way you format the items but don't allocate into String as part of the measurement.

File: components/calendar/examples/iso_date_manipulations.rs (hunk 1 above)
Comment (andrewpollack) [reply components/calendar/examples/iso_date_manipulations.rs]: Good call -- pushing update to do this now

**Review by Manishearth — APPROVED** [review]

**Pull request merged**